### PR TITLE
oneAPI 2024.1.0

### PR DESCRIPTION
**Pull request by pbchekin**
Build a new runner image 0.0.11 with oneAPI 2024.1.0 and use it for all workflows.

### CODE REVIEW

File: .github/workflows/e2e-performance.yml [hunk 1]
```diff
@@ -98,7 +98,7 @@ jobs:
     runs-on:
       - glados
       - spr
-      - runner-0.0.7
+      - runner-0.0.11
```
Comment (leshikus) [inline .github/workflows/e2e-performance.yml:101]: good catch, I've missed it

File: .github/actions/setup-ipex/action.yml [hunk 2]
```diff
@@ -33,9 +33,12 @@ runs:
     - name: Load IPEX wheels from a cache
       id: ipex-cache
       uses: ./.github/actions/load
+      env:
+        # Increase this value to reset cache
+        CACHE_NUMBER: "1"
```
Comment (leshikus) [inline .github/actions/setup-ipex/action.yml:38]: I wonder if Gregory's action `install-wheels` can be used instead

**Review by leshikus — APPROVED** [review]
LGTM

File: .github/actions/setup-ipex/action.yml (hunk 2 above)
Comment (pbchekin) [reply .github/actions/setup-ipex/action.yml:38]: action "install-wheels" loads wheels from artifacts, this one reads from cache.

**Pull request merged**